### PR TITLE
C# Cloud Spanner to 80% code coverage

### PR DESCRIPTION
**Pull request by benwulfe**
Runtime changes:
1. Fix for SpannerDbType.HashCode (bug for structs)
2. new SpannerParameterCollection.Clone along with usage in SpannerCommand.Clone
3. Code removal/cleanup
   Removal of Precondition (inline single usage)
   Remove duplicate code in synchronous OpenAsReadOnly
   Remove duplicate functionality for GetOridinal/GetFieldIndexAsync (prefer one that caches in a dictionary)
4. Extra tests for
   VolatileResourceManager
   SpannerDataReader.Get[Type] methods
   SpannerCommand
   SpannerDbType
   SpannerParameterCollection
   SpannerParameter
   SpannerProviderFactory
5. Fix for high stress test (retry needs to recreate the transaction).

### CODE REVIEW

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs [hunk 1]
```diff
@@ -59,6 +60,21 @@ public BindingTests(TestDatabaseFixture testFixture, ITestOutputHelper outputHel
...
+                        // we are disabling culture checks since we do both the same way (and valueRead is a T and cannot be fed culture).
+                        // ReSharper disable once SpecifyACultureInStringConversionExplicitly
+                        if (typeof(T) == typeof(decimal)) { Assert.Equal(reader.GetDecimal(0).ToString(), valueRead.ToString()); }
+                        // ReSharper disable once SpecifyACultureInStringConversionExplicitly
+                        if (typeof(T) == typeof(double)) { Assert.Equal(reader.GetDouble(0).ToString(), valueRead.ToString()); }
+                        // ReSharper disable once SpecifyACultureInStringConversionExplicitly
+                        if (typeof(T) == typeof(float)) { Assert.Equal(reader.GetFloat(0).ToString(), valueRead.ToString()); }
+                        if (typeof(T) == typeof(Int16)) { Assert.Equal(reader.GetInt16(0).ToString(), valueRead.ToString()); }
+                        if (typeof(T) == typeof(Int32)) { Assert.Equal(reader.GetInt32(0).ToString(), valueRead.ToString()); }
+                        if (typeof(T) == typeof(Int64)) { Assert.Equal(reader.GetInt64(0).ToString(), valueRead.ToString()); }
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs]: Only one test verifies each of these types right? What if `TestBind<T>` returns `Task<T>` instead and the callers verify this so you don't need all the type checks.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs [hunk 2]
```diff
@@ -467,6 +467,33 @@ await connection.BeginReadOnlyTransactionAsync(TimestampBound.Strong))
...
+        {
+            var writeCommand = writeConnection.CreateUpdateCommand(
+                "TX",
+                new SpannerParameterCollection
+                {
+                    {"k", SpannerDbType.String, _key},
+                    {"Int64Value", SpannerDbType.Int64, 0}
+                });
+            await writeCommand.ExecuteNonQueryAsync().ConfigureAwait(false);
+        }
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs]: Nit: add newline

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs [hunk 3]
```diff
@@ -505,6 +532,94 @@ await connection.BeginReadOnlyTransactionAsync(TimestampBound.Strong))
...
         }
+
+        [Fact]
+        public async Task ScopeCompleteWithTwoReadsDoesntThrow()
+        {
+            await WriteSampleRowsAsync();
+            using (var scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
+            {
+                var readConnection1 = await _testFixture.GetTestDatabaseConnectionAsync();
+                var readConnection2 = await _testFixture.GetTestDatabaseConnectionAsync();
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs]: Usings for these? (same comment in other 2 tests below). If you're worried about nesting by the way, you can stack them:
```csharp
using (var scope = new TransactionScope(TransactionScopeAsyncFlowOption.Enabled))
using (var readConnection1 = await _testFixture.GetTestDatabaseConnectionAsync())
using (var readConnection2 = await _testFixture.GetTestDatabaseConnectionAsync())
{
}
```

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+using System.Data;
+using System.Linq;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class SpannerCommandTests
+    {
+        [Fact]
+        public void OnlyCommandText()
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs]: Should this be OnlyCommand**Type**?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+        public void UpdateRowSourceNotSupported()
+        {
+            var command = new SpannerCommand();
+            Assert.Throws<NotSupportedException>(() => command.UpdatedRowSource = UpdateRowSource.Both);
+            Assert.Throws<NotSupportedException>(() => command.UpdatedRowSource = UpdateRowSource.FirstReturnedRecord);
+            Assert.Throws<NotSupportedException>(() => command.UpdatedRowSource = UpdateRowSource.OutputParameters);
+        }
+
+        [Fact]
+        public void PrePareNoop()
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs]: "P" => "p"

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                "T", new SpannerParameterCollection
+                {
+                    {"P1", SpannerDbType.String},
+                    {"P2", SpannerDbType.Float64}
+                });
+
+            var command2 = (SpannerCommand)command.Clone();
+            Assert.Same(command.SpannerConnection, command2.SpannerConnection);
+            Assert.Equal(command.CommandText, command2.CommandText);
+            Assert.Equal(command.CommandText, command2.CommandText);
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs]: Duplicate check

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+            Assert.Equal(command.Parameters.Count, command2.Parameters.Count);
+            Assert.NotSame(command.Parameters, command2.Parameters);
+            Assert.True(command.Parameters.All(x => command2.Parameters.Contains(x)));
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs]: Do the parameters have a consistent order? Can you use `SequenceEquals` here?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerDbTypeTests.cs [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class SpannerDbTypetests
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerDbTypeTests.cs]: Nit: SpannerDbType**T**ests

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerDbTypeTests.cs [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    {
+        public static IEnumerable<object[]> GetSpannerDbTypes()
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerDbTypeTests.cs]: Make this a local function in `EqualsTest` perhaps? If not, then make it private.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterCollectionTests.cs [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+        [Fact]
+        public void CopyToArray()
+        {
+            var parameters1 = Enumerable.Range(0, 5).Select(
+                x => new SpannerParameter($"Param{x}", SpannerDbType.String)).ToArray();
+            var parameters2 = new SpannerParameter[parameters1.Length];
+
+            var collection = new SpannerParameterCollection();
+            collection.AddRange(parameters1);
+            collection.CopyTo(parameters2, 0);
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterCollectionTests.cs]: You might also want to test copying to an array and not starting at index 0.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+using System.Collections.Generic;
+using System.Data;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class SpannerParameterTests
+    {
+        public static IEnumerable<object[]> GetDbTypeConversions()
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs]: Same comment: function local or private (if possible...I realize now this may be needed for reflection, so it might not be possible).

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+        {
+            yield return new object[] { SpannerDbType.Bytes, DbType.Binary};
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs]: Nit: space before closing brace.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+            yield return new object[] { SpannerDbType.Unspecified, DbType.Object };
+            yield return new object[] { SpannerDbType.String, DbType.String };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetDbTypeConversions))]
+        public void DbTypeMappgings(SpannerDbType spannerType, DbType adoType)
+        {
+            var parameter = new SpannerParameter();
+            parameter.SpannerDbType = spannerType;
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs]: Nit: `var parameter = new SpannerParameter { SpannerDbType = spannerType }` perhaps?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameterCollection.cs [hunk 14]
```diff
@@ -29,6 +29,23 @@ namespace Google.Cloud.Spanner.Data
...
+        public SpannerParameterCollection()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new SpannerParameterCollection with the given set of initial <see cref="SpannerParameter"/>
+        /// objects.
+        /// </summary>
+        /// <param name="initialParameters">The initial parameters to include in the collection.</param>
+        public SpannerParameterCollection(IEnumerable<SpannerParameter> initialParameters)
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerParameterCollection.cs]: Make this default to null perhaps and then just add a null check below? Then only one constructor is needed.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs [hunk 15]
```diff
@@ -233,8 +233,7 @@ public override int GetHashCode()
         {
             unchecked
             {
-                int hashCode = StructMembers?.GetHashCode() ?? 0;
-                hashCode = (hashCode * 397) ^ (int) TypeCode;
+                int hashCode = (int) TypeCode;
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs]: What was the bug here? Why is StructMembers not included anymore if it is still compared in `Equals`?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs (hunk 1 above)
Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs]: because at that time the spannerdatareader is shut down and passing around the datareader defeats the purpose of having a good helper function.
There are a finite number of type specific reader calls, so I don't think its a big deal to conditionally test them based on the inputs.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs (hunk 3 above)
Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs]: that cleans things up quite nicely.  thanks for that.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs (hunk 4 above)
Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs]: its sort of correct as is, but I made it clearer as:

CommandTypeSupportsOnlyCommandText

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs (hunk 15 above)
Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs]: two objects can have the same hashcode yet not be equal, but if they are equal, their hashcodes must be the same.
The problem is the struct dictionary -- where I am manually checking for equality by scanning contents, but was incorrectly assuming the hashcodes of two dictionaries would be the same if their contents were the same.
This would decrease perf slightly for dictionaries keyed off of SpannerDbType that had a lot of structs (due to having many colliding hashcodes) -- but that should be a pretty narrow use case if anyone ever actually does it.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs (hunk 1 above)
Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs]: Perhaps a better solution would be to take a `Func<SpannerDataReader, T> typeSpecificGetter` as an optional parameter, and then do:

```
if(typeSpecificGetter != null)
{
    Assert.Equal(valueRead, typeSpecificGetter(reader));
}
```

Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs]: I like it...

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs [hunk 16]
```diff
@@ -68,7 +68,10 @@ public SpannerCommand()
         {
             SpannerConnection = connection;
             _transaction = transaction;
-            Parameters = parameters;
+            if (parameters != null)
+            {
+                Parameters = parameters;
+            }
```
Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs]: Instead of initializing in two spots and adding an if, why not just null coalesce? 

`Parameters = parameters ?? new SpannerParameterCollection();`

That will keep the compile error if you create a new constructor that forgets to initialize.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs [hunk 17]
```diff
@@ -198,7 +201,7 @@ protected override DbTransaction DbTransaction
         /// Returns a copy of this <see cref="SpannerCommand"/>.
         /// </summary>
         /// <returns>a copy of this <see cref="SpannerCommand"/>.</returns>
-        public object Clone() => new SpannerCommand(SpannerConnection, _transaction, Parameters)
+        public object Clone() => new SpannerCommand(SpannerConnection, _transaction, Parameters?.Clone())
```
Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs]: I would be surprised if `Parameters` could ever be null.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+            yield return new object[] { SpannerDbType.Timestamp, DbType.DateTime };
+            yield return new object[] { SpannerDbType.Float64, DbType.Double };
+            yield return new object[] { SpannerDbType.Int64, DbType.Int64 };
+            yield return new object[] { SpannerDbType.Unspecified, DbType.Object };
+            yield return new object[] { SpannerDbType.String, DbType.String };
+        }
+
+        [Theory]
+        [MemberData(nameof(GetDbTypeConversions))]
+        public void DbTypeMappgings(SpannerDbType spannerType, DbType adoType)
```
Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerParameterTests.cs]: Strictly speaking, this should be two tests, but it is short enough that is doesn't really matter.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs [hunk 19]
```diff
@@ -0,0 +1,85 @@
...
+namespace Google.Cloud.Spanner.Data.Tests
+{
+    public class SpannerCommandTests
+    {
+        [Fact]
+        public void CommandTypeSupportsOnlyCommandText()
+        {
+            var command = new SpannerCommand();
+            Assert.Throws<NotSupportedException>(() => command.CommandType = CommandType.StoredProcedure);
+            Assert.Throws<NotSupportedException>(() => command.CommandType = CommandType.TableDirect);
```
Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs]: Also test that setting to `CommandText` does not throw.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs [hunk 20]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        [Fact]
+        public void UpdateRowSourceNotSupported()
+        {
+            var command = new SpannerCommand();
+            Assert.Throws<NotSupportedException>(() => command.UpdatedRowSource = UpdateRowSource.Both);
+            Assert.Throws<NotSupportedException>(() => command.UpdatedRowSource = UpdateRowSource.FirstReturnedRecord);
+            Assert.Throws<NotSupportedException>(() => command.UpdatedRowSource = UpdateRowSource.OutputParameters);
```
Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs]: Also test  setting to `None` does not throw.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs [hunk 21]
```diff
@@ -41,7 +41,7 @@ public BindingTests(TestDatabaseFixture testFixture, ITestOutputHelper outputHel
 
         private readonly TestDatabaseFixture _testFixture;
 
-        private async Task TestBind<T>(SpannerDbType parameterType, T value)
+        private async Task TestBind<T>(SpannerDbType parameterType, T value, Func<SpannerDataReader, T> typeSpecficReader = null)
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/BindingTests.cs]: Nit: "typeSpecficReader" => "typeSpecificReader"

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs [hunk 22]
```diff
@@ -0,0 +1,85 @@
...
+        }
+
+        [Fact]
+        public void PrepareNoop()
+        {
+            var command = new SpannerCommand();
+            command.Prepare();
```
Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs]: Perhaps assert the initial conditions are still true?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs [hunk 23]
```diff
@@ -505,6 +533,67 @@ await connection.BeginReadOnlyTransactionAsync(TimestampBound.Strong))
...
+        }
+
+        [Fact]
+        public async Task ScopeCompleteWithOneReadOneWriteThrows()
+        {
+            await WriteSampleRowsAsync();
+            await Assert.ThrowsAsync<TransactionAbortedException>(
+                async () =>
+                {
+
```
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs]: Nit: extra newline

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs (hunk 15 above)
Comment (evildour) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs]: Oh I see. Missed that it was a dictionary. Perhaps maybe just including its Count in the hash code then so you get a bit better hash code distribution?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs (hunk 17 above)
Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs]: it was before, so I'm being defensive here.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs (hunk 15 above)
Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs]: not a terrible idea...

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs [hunk 24]
```diff
@@ -467,6 +467,34 @@ await connection.BeginReadOnlyTransactionAsync(TimestampBound.Strong))
         }
 
 #if NET45 || NET451 || NET452
+        private async Task UpdateValueInTransactionScope(SpannerConnection writeConnection)
+        {
+            var writeCommand = writeConnection.CreateUpdateCommand(
+                "TX",
+                new SpannerParameterCollection
```
Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/TransactionTests.cs]: six/half dozen.  I prefer this style.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs (hunk 22 above)
Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs]: nah I could check all state, but its a bit overkill.  prepare is actually a no-op and is ignored.

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs (hunk 16 above)
Comment (benwulfe) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs]: how is that a compile error?  It will be just null in that case.  seems this way may be safer in that it will always be set when a new ctor is made.

**Comment by evildour** [comment]
LGTM. With these large PRs it's very difficult to see what has changed after addressing review comments. In the future, would it be possible to create new commits to address comments instead of amending the original?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs (hunk 16 above)
Comment (ILMTitan) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerCommand.cs]: It is not, I don't know what I was thinking.

**Comment by benwulfe** [comment]
sorry mike.  I'll squash them after an LGTM next time.  thanks for the reviews jim+mike.
